src/main/java/org/opentripplanner/routing/core/State.java

```diff
@@ -436,12 +302,10 @@ public TraverseMode getNonTransitMode() {
     // TODO: There is no documentation about what this means. No one knows precisely.
     // Needs to be replaced with clearly defined fields.
 
-    public State reversedClone() {
+    private State reversedClone() {
```

Comment:
Do we still need the state / path reversal logic at all? I guess this is used in the egress search but there might be a simpler way to do it now that we're no longer "reverse optimizing".

Reply:
I think it is used for arrive-by searches. If we assume that street searches are time-independent, we could just achieve arrive-by by time-shifting the result. We are making this assumption with the Raptor search now, but we will not be able to add things like traffic updates or roads that are only open at certain times of day.

In any case, we could probably clean up the reversal logic a little, if we decide to keep it.

Reply:
We decided to keep it for now, maybe try to eliminate later.